bindsnet/learning/reward.py

```diff
@@ -4,31 +4,28 @@
 import torch
 
 
-class AbstractRPE(ABC):
+class AbstractReward(ABC):
     # language=rst
     """
-    Abstract base class for reward prediction error (RPE).
+    Abstract base class for reward computation.
     """
 
     def __init__(self):
```

Comment:
You mean removing the `__init__`? I put the variables here because, depending on what your reward function is, you might want to plot extra stuff (in `Pipeline`), and we would then need to ensure that these attributes exist. I agree that the naming is incorrect though ATM. Or is there any other way we could achieve this?

Reply:
You could either remove `__init__` or just make it `pass`. I would put the variables in the `__init__` of the subclass(es) that need them. As for plotting, I think this can be done at the script level; that is, outside of the package code, so I don't really think that's a concern.

Reply:
You're right. I think I'll resort to making my own `Pipeline`-like object to allow some more freedom :smile: